src/GafferScene/EditScopeAlgo.cpp

```diff
@@ -854,3 +855,210 @@ const Gaffer::GraphComponent *GafferScene::EditScopeAlgo::attributeEditReadOnlyR
 
 	return nullptr;
 }
+
+// Set Membership
+// ==============
+
+namespace
+{
+
+static int g_addSetColumnIndex = 0;
+static int g_removeSetColumnIndex = 1;
+
+SceneProcessorPtr setMembershipProcessor()
+{
+	SceneProcessorPtr result = new SceneProcessor( "SetMembershipEdits" );
+
+	SpreadsheetPtr spreadsheet = new Spreadsheet;
+	result->addChild( spreadsheet );
+	spreadsheet->selectorPlug()->setValue( "${setMembership:set}" );
+	spreadsheet->rowsPlug()->addColumn( new StringVectorDataPlug( "add", Plug::Direction::In, new StringVectorData() ) );
+	spreadsheet->rowsPlug()->addColumn( new StringVectorDataPlug( "remove", Plug::Direction::In, new StringVectorData() ) );
+
+	PathFilterPtr addPathFilter = new PathFilter;
+	result->addChild( addPathFilter );
+	addPathFilter->pathsPlug()->setInput( spreadsheet->outPlug()->getChild<StringVectorDataPlug>( g_addSetColumnIndex ) );
+
+	PathFilterPtr removePathFilter = new PathFilter;
+	result->addChild( removePathFilter );
+	removePathFilter->pathsPlug()->setInput( spreadsheet->outPlug()->getChild<StringVectorDataPlug>( g_removeSetColumnIndex ) );
+
+	GafferScene::SetPtr addSet = new GafferScene::Set();
+	result->addChild( addSet );
+	addSet->inPlug()->setInput( result->inPlug() );
+	addSet->filterPlug()->setInput( addPathFilter->outPlug() );
+	addSet->namePlug()->setInput( spreadsheet->enabledRowNamesPlug() );
+	addSet->modePlug()->setValue( GafferScene::Set::Mode::Add );
+	addSet->setVariablePlug()->setValue( "setMembership:set" );
+
+	GafferScene::SetPtr removeSet = new GafferScene::Set();
+	result->addChild( removeSet );
+	removeSet->inPlug()->setInput( addSet->outPlug() );
+	removeSet->filterPlug()->setInput( removePathFilter->outPlug() );
+	removeSet->namePlug()->setInput( spreadsheet->enabledRowNamesPlug() );
+	removeSet->modePlug()->setValue( GafferScene::Set::Mode::Remove );
+	removeSet->setVariablePlug()->setValue( "setMembership:set" );
+
+	auto rowsPlug = static_cast<Spreadsheet::RowsPlug *>(
+		PlugAlgo::promoteWithName( spreadsheet->rowsPlug(), "edits" )
+	);
+
+	Metadata::registerValue( rowsPlug, "spreadsheet:defaultRowVisible", new BoolData( false ) );
+	Metadata::registerValue( rowsPlug->defaultRow(), "spreadsheet:rowNameWidth", new IntData( 300 ) );
```

Comment:
The other processors probably use this width to accommodate long paths in the row names. Since we're dealing with sets in the row names, we can probably make this slimmer, but should perhaps widen the columns?

Reply:
Fixed in 7aa071c6c8c5daef2546b55d885b115d5ef17358